homeassistant/components/transmission.py

```diff
@@ -0,0 +1,171 @@
+"""
+Component for monitoring the Transmission BitTorrent client API.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/transmission/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD, CONF_NAME,
+                                 CONF_PORT, CONF_USERNAME,
+                                 CONF_MONITORED_VARIABLES)
+
+REQUIREMENTS = ['transmissionrpc==0.11']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'transmission'
+DATA_TRANSMISSION = 'TRANSMISSION'
+
+DEFAULT_NAME = 'Transmission'
+DEFAULT_PORT = 9091
+
+SENSOR_TYPES = {
+    'active_torrents': ['Active Torrents', None],
+    'current_status': ['Status', None],
+    'download_speed': ['Down Speed', 'MB/s'],
+    'paused_torrents': ['Paused Torrents', None],
+    'total_torrents': ['Total Torrents', None],
+    'upload_speed': ['Up Speed', 'MB/s'],
+    'completed_torrents': ['Completed Torrents', None],
+    'started_torrents': ['Started Torrents', None],
+}
+
+CONFIG_SCHEMA = vol.Schema({
+  DOMAIN: vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=['current_status']):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+  })
+}, extra=vol.ALLOW_EXTRA)
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+def setup(hass, config):
+    hass.data[DATA_TRANSMISSION] = TransmissionData(hass, config)
+
+    def refresh(event_time):
+        hass.data[DATA_TRANSMISSION].update()
+
+    track_time_interval(hass, refresh, SCAN_INTERVAL)
+
+    sensorconfig = {
+        'sensors': config[DOMAIN].get(CONF_MONITORED_VARIABLES),
+        'client_name': config[DOMAIN].get(CONF_NAME)}
+    load_platform(hass, 'sensor', DOMAIN, sensorconfig)
+
+    _LOGGER.info("Transmission component setup completed.")
+    return True
+
+
+class TransmissionData:
+    """Get the latest data and update the states."""
+
+    def __init__(self, hass, config):
+        import transmissionrpc
+        from transmissionrpc.error import TransmissionError
+        try:
+            """Initialize the Transmission RPC API"""
+            host = config[DOMAIN].get(CONF_HOST)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.